File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    heapq.heapify(scoville)
    answer=0
    while(True):
        a=heapq.heappop(scoville)
        if(a>=K):
            break
        if(len(scoville)==0):
            answer=-1
            break
        b=heapq.heappop(scoville)
        heapq.heappush(scoville,a+b*2)
        answer+=1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        Queue<Integer> pq = new PriorityQueue<>((a, b) -> {
            return a - b;
        });
        for(int sco : scoville){
            pq.offer(sco);
        }
        int[] plus = new int[2];
        int cur = 0;
        while(pq.size() >= 2 && pq.peek() < K){
            plus[0]= pq.poll();
            plus[1]= pq.poll();
            cur= plus[0] + 2* plus[1];
            pq.offer(cur);
            answer++;
        }
        if(pq.peek() < K){
            answer= -1;
        }
        return answer;
    }
}